sdk/data_cosmos/src/clients/cosmos.rs

```diff
@@ -154,28 +148,24 @@ fn new_pipeline_from_options(
 }
 
 /// The cloud with which you want to interact.
-///
-/// All variants require the cosmos account name. `Custom` also requires a valid
-/// base URL (e.g. https://custom.documents.azure.com)
-#[allow(unused)]
+// TODO: Other govt clouds?
 #[derive(Debug, Clone)]
-enum CloudLocation {
+pub enum CloudLocation {
     /// Azure public cloud
-    Public(String),
+    Public { account: String },
     /// Azure China cloud
-    China(String),
-    // TODO: Other govt clouds?
+    China { account: String },
     /// A custom base URL
-    Custom { account: String, uri: String },
+    Custom { uri: String },
 }
 
 impl CloudLocation {
     /// the base URL for a given cloud location
     fn url(&self) -> String {
         match self {
-            CloudLocation::Public(account) => format!("https://{}.documents.azure.com", account),
-            CloudLocation::China(account) => format!("https://{}.documents.azure.cn", account),
-            CloudLocation::Custom { uri, .. } => uri.clone(),
+            CloudLocation::Public { account } => format!("https://{}.documents.azure.com", account),
```

Comment:
If you run `az cloud list`. This is not listed there, but you logically be in the `suffixes`.
``` json
    "suffixes": {
      "acrLoginServerEndpoint": ".azurecr.io",
      "attestationEndpoint": ".attest.azure.net",
      "azureDatalakeAnalyticsCatalogAndJobEndpoint": "azuredatalakeanalytics.net",
      "azureDatalakeStoreFileSystemEndpoint": "azuredatalakestore.net",
      "keyvaultDns": ".vault.azure.net",
      "mariadbServerEndpoint": ".mariadb.database.azure.com",
      "mhsmDns": ".managedhsm.azure.net",
      "mysqlServerEndpoint": ".mysql.database.azure.com",
      "postgresqlServerEndpoint": ".postgres.database.azure.com",
      "sqlServerHostname": ".database.windows.net",
      "storageEndpoint": "core.windows.net",
      "storageSyncEndpoint": "afs.azure.net",
      "synapseAnalyticsEndpoint": ".dev.azuresynapse.net"
    }
```